api/models.py

```diff
@@ -116,6 +128,9 @@ class User(DatabaseModel):
         default=False,
         description="True if superuser otherwise False"
     )
+    groups: Optional[List[UserGroup]] = Field(
```

Comment:
I think it is better to just keep it `null` rather than assigning an empty list if the user belongs to no group.

Reply:
Why do you think it's better?

Making it not optional means the field will always have to be of the same type, so we can do things like `if group in user.groups` without checking whether `user.groups` is None or a list.  IWe can also set `default=[]` so the user object could be created without passing a list of groups (the client code in kernelci-core should probably be passing an empty list by default anyway, but that's an implementation detail).

About having it optional, the only real reason I could think of would be to make it backwards-compatible if we had already published a schema without `.groups`.  But that's not the case, and even then it would still not break things with a default value.

Reply:
The reason is just it is simpler to not set anything if it doesn't have any group rather than setting it with an empty value.
Yes, I am aware that we can make it backward-compatible by setting `default` field value.
Also, we can check `if user.groups` before iterating over groups if we keep it `Optional`.
I don't see much of a difference here. These are just 2 different approaches we can go with.

Reply:
I think it is better to set the default value and drop `Optional` to make further comparison easier to authorize users in POST node requests.